### PR TITLE
WIP: Adding maven_deploy_artifact job; maven cache/restore can use variable pom_file

### PR DESCRIPTION
**Pull request by cesarfm**

### CODE REVIEW

File: src/commands/maven_restore_artifacts.yml [hunk 1]
```diff
@@ -11,13 +11,17 @@ parameters:
...
 
 steps:
   - when:
       condition: << parameters.prefix >>
       steps:
         - restore_cache:
             keys:
-              - << parameters.prefix >>-{{ checksum "pom.xml" }}-<< parameters.path >>
-              - << parameters.prefix >>-{{ checksum "pom.xml" }}
+              - << parameters.prefix >>-{{ checksum "<< parameters.pom_file >>" }}-<< parameters.path >>
```
Comment (cesarfm) [inline src/commands/maven_restore_artifacts.yml:25]: In order to cover https://github.com/ricardo-ch/ricardo-spring-starters/pull/151#discussion_r952225457

File: src/commands/maven_cache_artifacts.yml [hunk 2]
```diff
@@ -10,9 +10,13 @@ parameters:
     description: 'Path of maven module for the app, or "." for single-app-repo'
     type: string
     default: '.'
+  pom_file:
+    description: 'The pom file to be used for the Maven build'
+    type: string
+    default: 'pom.xml'
```
Comment (michaelkuechler) [inline src/commands/maven_cache_artifacts.yml:16]: add param to readme

File: src/commands/maven_deploy_artifact.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+parameters:
+  path:
+    description: 'Path of maven module for the app, or "." for single-app-repo'
+    type: string
+    default: .
+  pom_file:
+    description: 'The pom file to be used for the Maven build'
+    type: string
+    default: 'pom.xml'
```
Comment (michaelkuechler) [inline src/commands/maven_deploy_artifact.yml:13]: add command to readme

File: src/commands/maven_restore_artifacts.yml [hunk 4]
```diff
@@ -11,13 +11,17 @@ parameters:
     description: 'Path of maven module for the app, or "." for single-app-repo'
     type: string
     default: '.'
+  pom_file:
+    description: 'The pom file to be used for the Maven build'
+    type: string
+    default: 'pom.xml'
```
Comment (michaelkuechler) [inline src/commands/maven_restore_artifacts.yml:17]: add to readme

File: src/jobs/maven_deploy_artifact.yml [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    default: ''
+  pom_file:
+    description: 'The pom file to be used for the Maven build'
+    type: string
+    default: 'pom.xml'
+  executor:
+    description: 'Executor for the build (java_builder_docker is more lightweight, but some tests leveraging testcontainers etc require the java_builder_vm)'
+    type: enum
+    enum: [ 'maven_docker', 'maven_vm' ]
+    default: 'maven_docker'
```
Comment (michaelkuechler) [inline src/jobs/maven_deploy_artifact.yml:22]: add job to readme

**Review by michaelkuechler — APPROVED** [review]
see comments

**Pull request merged**